### PR TITLE
Bump `:gun` to 2.0.0-rc.1.

### LINKED ISSUE #236: Update (or relax) dependencies
Hello everyone! Just tried the lib today by creating a brand new Phoenix app and I got the following error:

```
Failed to use "cowlib" (version 2.9.1) because
  cowboy (version 2.8.0) requires ~> 2.9.1
  gun (versions 1.3.0 to 1.3.2) requires ~> 2.6.0
  mix.lock specifies 2.9.1


Failed to use "cowlib" (version 2.9.1) because
  cowboy (version 2.8.0) requires ~> 2.9.1
  gun (version 1.3.3) requires ~> 2.7.0
  mix.lock specifies 2.9.1
```

Do you think we could have this updated?
 
PS.: Out of curiosity: Is the `gun` lib really necessary if using `httpoison`?

### PR DESCRIPTION
**Pull request by jchristgit**
This prevents dependency problems when trying to use nostrum with e.g. Phoenix,
as outlined in #236. It also sets the foundation for using `:gun` as our one
and only HTTP client, instead of relying on an extra library here.

Fixes #236.

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -91,7 +91,7 @@ defmodule Nostrum.Mixfile do
     [
       {:httpoison, "~> 1.7"},
       {:poison, "~> 3.0"},
-      {:gun, "~> 1.3"},
+      {:gun, github: "ninenines/gun", tag: "2.0.0-rc.1"},
```
Comment (Kraigie) [inline mix.exs:94]: I'm not sure we can do a hex release with a GH dependency. If we can't we should try and see if one of the other available versions from hex would work. It looks like hex is currently hosting the rc version you have here.

**Review by Kraigie — APPROVED** [review]
One small comment we should probably look into, LGTM otherwise!

File: mix.exs (hunk 1 above)
Comment (jchristgit) [reply mix.exs]: Craig, you are absolutely right. I've pinned this to the hex pre-release instead.

**Comment by jchristgit** [comment]
Thanks!

**Pull request merged**